tools/generators/ethereum/contract_parsing.go

```diff
@@ -218,22 +220,46 @@ func buildMethodInfo(
 			params += fmt.Sprintf("%v,\n", paramName)
 
 			parsingFn := ""
-			switch param.Type.String() {
-			case "bytes":
-				parsingFn = "hexutil.Decode"
-			case "address":
-				parsingFn = "chainutil.AddressFromHex"
-			case "uint256":
-				parsingFn = "hexutil.DecodeBig"
-			default:
-				commandCallable = false
+
+			paramStructured := false
+			if param.Type.TupleType != nil {
+				paramName += "_json"
+				paramStructured = true
+			} else {
+				switch goType {
+				case "[]byte":
+					parsingFn = "hexutil.Decode(%s)"
+				case "common.Address":
+					parsingFn = "chainutil.AddressFromHex(%s)"
+				case "*big.Int":
+					parsingFn = "hexutil.DecodeBig(%s)"
+				case "bool":
+					parsingFn = "strconv.ParseBool(%s)"
+				case "uint64":
```

Comment:
It's not as easy as you think, because `strconv.ParseUint(width, 10, 32)` will return `uint64` that we would need to cast to `uint32` or `uint8`. Can we do it in a separate PR?

Reply:
5e55a13fb7e761d73017947ec049e3d2b23e65b5